manager/scheduler/scheduler_test.go

```diff
@@ -2098,6 +2167,42 @@ func TestSchedulerPluginConstraint(t *testing.T) {
 	assignment2 := watchAssignment(t, watch)
 	assert.Equal(t, assignment2.ID, "task3_ID")
 	assert.Equal(t, assignment2.NodeID, "node3_ID")
+
+	// Create t4; it should stay in the pending state because there is
+	// no node that with log plugin `plugin1`
+	err = s.Update(func(tx store.Tx) error {
+		assert.NoError(t, store.CreateTask(tx, t4))
+		return nil
+	})
+	assert.NoError(t, err)
+
+	time.Sleep(100 * time.Millisecond)
+	s.View(func(tx store.ReadTx) {
+		task := store.GetTask(tx, "task4_ID")
+		if task.Status.State >= api.TaskStateAssigned {
+			t.Fatal("task 'task4_ID' should not have been assigned")
+		}
+	})
```

Comment:
There is a better way to do this - though it looks like this test hasn't been updated. You can use `watchAssignmentFailure`. For example:

```go
        failure := watchAssignmentFailure(t, watch)
        assert.Equal(t, "no suitable node (host-mode port already in use on 2 nodes)", failure.Status.Message)
```

If you have time to update the 3 existing places in this file with the 100 ms sleep, that would be really great, but it's also okay to merge it like this and I can update the tests later.